plugin.audio.pureradio/resources/lib/addon.py

```diff
@@ -0,0 +1,100 @@
+# -*- coding: utf-8 -*-
+
+'''
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+'''
+
+from resources.lib.plugin import Plugin
+import resources.lib.api as api
+
+plugin_id = 'plugin.audio.pureradio'
+
+p = Plugin(plugin_id)
+
+settings = p.get_plugin_settings()
+translation = p.get_plugin_translation()
+
+localized_strings =  {
+        'Play'     : 30010,
+        'Podcasts' : 30011,
+}
+
+# Entry point
+def run():
+    """This function is the entry point to the add-on.
+    It gets the add-on parameters and call the 'action' function.
+    """
+    p.log("pureradio.run")
+    # Get the params
+    params = p.get_plugin_parms()
+    action = params.get("action", '')
+    if action:
+        eval("%s(params)" % action)
```

Comment:
You should avoid using eval calls, especially in entry points : https://nedbatchelder.com/blog/201206/eval_really_is_dangerous.html

Reply:
replaced eval with static function calls